internal/deprecation/deprecation.go

```diff
@@ -130,12 +131,15 @@ func (checker *Checker) check(versionNumber string) (*Info, *failures.Failure) {
 }
 
 func (checker *Checker) shouldFetch() bool {
+	if !fileutils.FileExists(checker.deprecationFile) {
+		return true
+	}
+
 	lastFetch := checker.config.GetTime(fetchKey)
 	if !lastFetch.IsZero() && time.Now().Before(lastFetch) {
 		return false
 	}
 
-	checker.config.Set(fetchKey, time.Now().Add(15*time.Minute))
```

Comment:
Rather than remove it here let's add it to the start with a defer. That way the behaviour remains centralized.

Reply:
I would prefer to have it centralized as well, however adding a defer will mean that the `fetchKey` is set even when this function returns `false`. I could defer an anonymous func but that would be making this function more messy. I could also just call the `Set` function in two places here if keeping it centralized is what we want.

Reply:
hmm ok. How about we invert the conditional and only do the lastCheck stuff if the file exists? That's nearly the same as your first iteration, I know, except that GetTime doesn't get called and the logic is a bit more legible.